cmake_modules/android_setup.cmake

```diff
@@ -0,0 +1,73 @@
+#  at: http://www.maidsafe.net/licenses                                                            #
+#                                                                                                  #
+#  Unless required by applicable law or agreed to in writing, the MaidSafe Software distributed    #
+#  under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF   #
+#  ANY KIND, either express or implied.                                                            #
+#                                                                                                  #
+#  See the Licences for the specific language governing permissions and limitations relating to    #
+#  use of the MaidSafe Software.                                                                   #
+#                                                                                                  #
+#==================================================================================================#
+#                                                                                                  #
+#  Module used to create android setup for each project                                            #
+#                                                                                                  #
+#==================================================================================================#
+
+# Disable android builds for any platform but linux
+if(MSVC OR APPLE)
+  message(FATAL_ERROR "Android builds are currently not supported in this platform")
+endif()
+
+if(NOT ANDROID_NDK_TOOLCHAIN_ROOT)
+  set(ErrorMessage "\n\nCan't find ANDROID_NDK_TOOLCHAIN_ROOT. ")
+  set(ErrorMessage "${ErrorMessage}To set it, run:\n")
+  set(ErrorMessage "${ErrorMessage}    cmake . -DANDROID_NDK_TOOLCHAIN_ROOT=\"<path to ndk toolchain>\"\n\n")
+  message(FATAL_ERROR "${ErrorMessage}")
+endif()
+
+function(get_android_api_version_from_file File ApiLevel)
+  if(EXISTS "${File}")
+    file(STRINGS "${File}" AndroidVersionLine REGEX "#define __ANDROID_API__")
+    string(REGEX MATCH "[0-9]+" MatchedVersion "${AndroidVersionLine}")
+    if(MatchedVersion)
+      set(${ApiLevel} ${MatchedVersion} PARENT_SCOPE)
+    else()
+      unset(${ApiLevel} PARENT_SCOPE)
+    endif()
+  endif()
+endfunction()
+
+set(CMAKE_C_COMPILER "${ANDROID_NDK_TOOLCHAIN_ROOT}/bin/arm-linux-androideabi-gcc")
+set(CMAKE_CXX_COMPILER "${ANDROID_NDK_TOOLCHAIN_ROOT}/bin/arm-linux-androideabi-g++")
+
+macro(ms_android_setup_flags)
+  get_android_api_version_from_file("${ANDROID_NDK_TOOLCHAIN_ROOT}/sysroot/usr/include/android/api-level.h" AndroidApiLevel)
+  string(REGEX REPLACE "([0-9]\\.[0-9]).*" "\\1" AndroidGccVersion "${CMAKE_CXX_COMPILER_VERSION}")
+
+  set(ExeLinkerFlags "-L${ANDROID_NDK_TOOLCHAIN_ROOT}/arm-linux-androideabi/lib/armv7-a -lstdc++")
+
+  if(AndroidApiLevel LESS 16)
+    set(PlatformFlags )
```

Comment:
Should `PlatformFlags` be set to something here?  If not, I guess we can replace the if/else block with just an if block

    if (NOT AndroidApiLevel LESS 16)
      ...
    endif()